src/arrayMethodJoin.js

```diff
@@ -4,8 +4,31 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let strJoin = '';
+
+    if (this.length === 0) {
+      return strJoin;
+    }
```

Comment:
```suggestion
```
No need for this check. If there are no elements in the array, nothing will happen. Your loop will not even start, and in the end you will still return an empty string either way